realm/realm-library/src/objectServer/java/io/realm/RealmFunctions.java

```diff
@@ -17,30 +17,173 @@
 
 import org.bson.codecs.configuration.CodecRegistry;
 
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.realm.internal.Util;
 import io.realm.internal.jni.JniBsonProtocol;
+import io.realm.internal.jni.OsJNIResultCallback;
+import io.realm.internal.objectstore.OsJavaNetworkTransport;
+
+/**
+ * A <i>Realm functions<i> manager to call MongoDB Realm functions.
+ *
+ * Arguments and results are encoded/decoded with the <i>Realm functions'</i> codec registry either
+ * inherited from the {@link RealmAppConfiguration#getDefaultCodecRegistry()} or set explicitly
+ * when creating the <i>Realm functions</i>-instance through {@link RealmUser#getFunctions(CodecRegistry)}
+ * or through the individual calls to {@link #callFunction(String, List, Class, CodecRegistry)}.
+ */
+// TODO Timeout is currently handled uniformly through OkHttpNetworkTransport configured through RealmAppConfig
+// FIXME Application wide invocation; like StitchAppClient appClient = Stitch.getDefaultAppClient();
+// FIXME Encoder/Decoder specific variants
+// FIXME Service?
+public class RealmFunctions {
```

Comment:
We should probably also expose `getApp()` and `getUser()` functions from this class. This is useful if the class is passed around by itself. We are doing something similar in `ApiKeyAuth`. This would also require tests.

Reply:
👍

Reply:
👍